packages/offix-datastore/src/predicates/Operators.ts

```diff
@@ -1,10 +1,45 @@
-const equalsOperator = (modelField: any, value: any) => modelField === value;
-const gtOperator = (modelField: any, value: any) => modelField > value;
+type ArrayOperators = "ne" | "eq" | "in" | "contains";
+type MathematicalOperators = "ne" | "eq" | "le" | "lt" | "ge" | "gt" | "in";
 
-// https://github.com/aerogear/OpenVolunteerPlatform/blob/master/platform/client/src/dataFacade.tsx
-// TODO ADD type safety
-// TODO align operators to graphqlcrud
-export const AllOperators: any = {
-    "eq": equalsOperator,
-    "gt": gtOperator
+/**
+ * All Operators
+ *
+ * 'ne' - Is value not equal to input
+ * 'eq' - Is value equal to input
+ * 'le' - Is value less than or equal to input
+ * 'lt' - Is value strictly less than input
+ * 'ge' - Is value greater than or equal to input
+ * 'gt' - Is value strictly greater than input
+ * 'in' - Does input array or string contain value
+ * 'contains' - Is the input contained in value(array or string)
+ * 'startsWith' - Does value start with input string
+ * 'endsWith' - Does value end with input string
+ */
+export type AllOperators = MathematicalOperators | "contains" | "startsWith" | "endsWith";
+
+/**
+ * Maps data type to allowed operators
+ */
+export type TypeOperatorMap<T> =
+    T extends string[] ? ArrayOperators :
+    T extends string ? AllOperators :
+    T extends number[] ? ArrayOperators:
+    T extends number ? MathematicalOperators :
+    T extends boolean[] ? ArrayOperators:
+    T extends boolean ? "ne" | "eq" | "in" :
+    T extends Date[] ? ArrayOperators:
+    T extends Date ? MathematicalOperators :
+    AllOperators;
+
+export const OperatorFunctionMap = {
+    ne: (m: any, v: any) => m !== v,
+    eq: (m: any, v: any) => m === v,
+    ge: (m: number | string | Date, v: number | string | Date) => m >= v,
```

Comment:
Really nice workaround for now. We can make this generic by accepting type itself. Something like here: https://github.com/aerogear/graphback/blob/master/packages/graphback-core/src/runtime/QueryFilter.ts#L82